### PR TITLE
Fix partial clearing of section, fixes some bugs

### PR DESCRIPTION
**Pull request by ostrolucky**
Hello @pierredup, I have fixed a bug I reported and covered it with unit tests. Can you merge please? Thanks!

ref: https://github.com/symfony/symfony/pull/24363#pullrequestreview-97426883

### CODE REVIEW

File: src/Symfony/Component/Console/Output/ConsoleSectionOutput.php [hunk 1]
```diff
@@ -55,11 +55,14 @@ public function clear(int $lines = null)
             return;
         }
 
-        if (!$lines) {
+        if ($lines) {
+            $newContent = implode(PHP_EOL, \array_slice(explode(PHP_EOL, $this->content), 0, -$lines - 1));
```
Comment (ostrolucky) [inline src/Symfony/Component/Console/Output/ConsoleSectionOutput.php:59]: @nicolas-grekas Can you please give advice if there is easy way to optimize this line?

Comment (nicolas-grekas) [reply src/Symfony/Component/Console/Output/ConsoleSectionOutput.php:59]: Not sure :)

Comment (ostrolucky) [reply src/Symfony/Component/Console/Output/ConsoleSectionOutput.php:59]: Ok thanks, just wanted to know if there isn't something obvious by the chance

**Comment by pierredup** [comment]
Thanks for the effort, I have gone with a different approach, but used your test cases. Thanks!

**Comment by ostrolucky** [comment]
Thanks, your approach is better :)

**Pull request closed without merging**